sklearn/tree/tests/test_tree.py

```diff
@@ -2494,30 +2531,35 @@ def test_missing_value_errors(sparse_container, tree):
         tree.fit(X, y)
 
 
-def test_missing_values_poisson():
+@pytest.mark.parametrize("Tree", REG_TREES.values())
+def test_missing_values_poisson(Tree):
     """Smoke test for poisson regression and missing values."""
     X, y = diabetes.data.copy(), diabetes.target
 
     # Set some values missing
     X[::5, 0] = np.nan
     X[::6, -1] = np.nan
 
-    reg = DecisionTreeRegressor(criterion="poisson", random_state=42)
+    reg = Tree(criterion="poisson", random_state=42)
     reg.fit(X, y)
 
     y_pred = reg.predict(X)
     assert (y_pred >= 0.0).all()
 
 
 @pytest.mark.parametrize(
-    "make_data, Tree",
+    "make_data, Tree, resilience_score, dummy_model",
     [
-        (datasets.make_regression, DecisionTreeRegressor),
-        (datasets.make_classification, DecisionTreeClassifier),
+        (datasets.make_regression, DecisionTreeRegressor, 0.9, DummyRegressor),
+        (datasets.make_classification, DecisionTreeClassifier, 0.9, DummyClassifier),
+        (datasets.make_regression, ExtraTreeRegressor, 0.6, DummyRegressor),
+        (datasets.make_classification, ExtraTreeClassifier, 0.6, DummyClassifier),
     ],
 )
 @pytest.mark.parametrize("sample_weight_train", [None, "ones"])
-def test_missing_values_is_resilience(make_data, Tree, sample_weight_train):
+def test_missing_values_is_resilience(
```

Comment:
In this unit-test, I added different sensitivities to missing-values since a greedy-approach will be more resilient to missing-values completely at random.